src/HexagonOffload.cpp

```diff
@@ -119,114 +125,119 @@ class InjectHexagonRpc : public IRMutator {
-            // or the loop itself? Currently, this moves the loop itself.
-            Closure c(body);
-
-            // Make an argument list, and generate a function in the
-            // device_code module. The hexagon runtime code expects
-            // the arguments to appear in the order of (input buffers,
-            // output buffers, input scalars).  Scalars must be last
-            // for the scalar arguments to shadow the symbols of the
-            // buffer that get generated by CodeGen_LLVM.
-            std::vector<LoweredArgument> input_buffers, output_buffers;
-            std::map<std::string, Parameter> replacement_params;
-            for (const auto& i : c.buffers) {
-                if (i.second.write) {
-                    Argument::Kind kind = Argument::OutputBuffer;
-                    output_buffers.push_back(LoweredArgument(i.first, kind, i.second.type, i.second.dimensions));
-                } else {
-                    Argument::Kind kind = Argument::InputBuffer;
-                    input_buffers.push_back(LoweredArgument(i.first, kind, i.second.type, i.second.dimensions));
-                }
-
-                // Build a parameter to replace.
-                Parameter p(i.second.type, true, i.second.dimensions);
-                // Assert that buffers are aligned to one HVX vector.
-                const int alignment = 128;
-                p.set_host_alignment(alignment);
-                // The other parameter constraints are already
-                // accounted for by the closure grabbing those
-                // arguments, so we only need to provide the host
-                // alignment.
-                replacement_params[i.first] = p;
-
-                // Add an assert to the body that validates the
-                // alignment of the buffer.
-                if (!device_code.target().has_feature(Target::NoAsserts)) {
-                    Expr host_ptr = reinterpret<uint64_t>(Variable::make(Handle(), i.first + ".host"));
-                    Expr error = Call::make(Int(32), "halide_error_unaligned_host_ptr",
-                                            {i.first, alignment}, Call::Extern);
-                    body = Block::make(AssertStmt::make(host_ptr % alignment == 0, error), body);
-                }
-            }
-            body = replace_params(body, replacement_params);
-
-            std::vector<LoweredArgument> args;
-            args.insert(args.end(), input_buffers.begin(), input_buffers.end());
-            args.insert(args.end(), output_buffers.begin(), output_buffers.end());
-            for (const auto& i : c.vars) {
-                LoweredArgument arg(i.first, Argument::InputScalar, i.second, 0);
-                if (alignment_info.contains(i.first)) {
-                    arg.alignment = alignment_info.get(i.first);
-                }
-                args.push_back(arg);
+        if (loop->device_api != DeviceAPI::Hexagon) {
```

Comment:
Agreed, sorry. FWIW, the reason I didn't think about this is because the body of the code should actually be unchanged, but I will fix this now.